vnext/Microsoft.ReactNative/ReactApplication.h

```diff
@@ -1,48 +1,123 @@
-#pragma once
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+#pragma once
 
 #include "ReactApplication.g.h"
 #include "ReactNativeHost.h"
 
-#include <winrt/Windows.ApplicationModel.Activation.h>
-#include <winrt/Windows.UI.Core.h>
-
 using namespace winrt;
 using namespace Windows::ApplicationModel;
 using namespace Windows::UI::Xaml::Navigation;
+using namespace Microsoft::ReactNative;
+using namespace Microsoft::ReactNative::Bridge;
+using namespace Windows::Foundation::Collections;
 
 namespace winrt::Microsoft::ReactNative::implementation {
+
 struct ReactApplication : ReactApplicationT<ReactApplication> {
-  ReactApplication();
+  ReactApplication() noexcept;
 
-  Microsoft::ReactNative::ReactNativeHost Host();
+  Microsoft::ReactNative::ReactInstanceSettings InstanceSettings() noexcept;
+  void InstanceSettings(
+      Microsoft::ReactNative::ReactInstanceSettings const &value) noexcept;
 
-  virtual Microsoft::ReactNative::ReactNativeHost HostCore();
+  IVector<IReactPackageProvider> PackageProviders() noexcept;
+  void PackageProviders(IVector<IReactPackageProvider> const &value) noexcept;
 
-  virtual void OnLaunched(Activation::LaunchActivatedEventArgs const &);
+  hstring MainComponentName() noexcept;
+  void MainComponentName(hstring const &value) noexcept;
 
-  virtual void OnSuspending(IInspectable const &, SuspendingEventArgs const &);
+  bool UseDeveloperSupport() noexcept;
+  void UseDeveloperSupport(bool value) noexcept;
+
+  hstring JavaScriptMainModuleName() noexcept;
+  void JavaScriptMainModuleName(hstring const &value) noexcept;
+
+  hstring JavaScriptBundleFile() noexcept;
+  void JavaScriptBundleFile(hstring const &value) noexcept;
 
+  Microsoft::ReactNative::ReactNativeHost Host() noexcept;
+
+  virtual void OnLaunched(Activation::LaunchActivatedEventArgs const &);
+  virtual void OnSuspending(IInspectable const &, SuspendingEventArgs const &);
   virtual void OnNavigationFailed(
       IInspectable const &,
       NavigationFailedEventArgs const &);
 
  protected:
-  virtual winrt::Microsoft::ReactNative::
-      ReactApplicationDelegate __stdcall CreateReactApplicationDelegate() {
-    return winrt::Microsoft::ReactNative::ReactApplicationDelegate(*this);
-  }
+  virtual ReactApplicationDelegate __stdcall CreateReactApplicationDelegate();
```

Comment:
>__stdcall [](start = 35, length = 9)

why is this necessary?

Reply:
I am not sure. It was there before. I would like to remove the ReactApplicationDelegate in future iterations. IMHO, there are too many types in the API that we must remove.